Assets/BossRoom/Scripts/Client/UI/PostGameUI.cs

```diff
@@ -77,18 +94,15 @@ void SetPostGameUI(WinState winState)
         public void OnPlayAgainClicked()
         {
             // this should only ever be called by the Host - so just go ahead and switch scenes
+            Assert.IsTrue(NetworkManager.Singleton.IsServer);
             SceneLoaderWrapper.Instance.LoadScene("CharSelect");
 
             // FUTURE: could be improved to better support a dedicated server architecture
         }
 
         public void OnMainMenuClicked()
         {
-            // Player is leaving this group - leave current network connection first
-            var gameNetPortal = GameObject.FindGameObjectWithTag("GameNetPortal").GetComponent<GameNetPortal>();
-            gameNetPortal.RequestDisconnect();
-
-            SceneLoaderWrapper.Instance.LoadScene("MainMenu");
+            m_ApplicationController.LeaveSession();
```

Comment:
This function was already doing the same as we did in the PostGameUI function but it also properly leaves the lobby.